### DISCUSSION TITLE
Nullable field support

### DISCUSSION THREAD

**Discussion by flaviovich** [prisma/prisma#18432] [Q&A] [answered]
Hi!
Does view support a nullable field in a view?
I'm getting this error:
> Error converting field "Respuesta" of expected non-nullable type "String", found incompatible value of "null".

This is my view:
```js
view RespuestasDetalle {
    DocId       String  @unique @db.VarChar(15)
    Colaborador String  @db.VarChar(45)
    Sede        String  @db.VarChar(45)
    Pregunta    Int     @db.UnsignedSmallInt
    Respuesta   String? @db.Char(1)
    Resultado   String  @db.VarChar(15)
}
```
Maybe it's because I added '?' later?

**Comment by ludralph** [comment]
Hi @flaviovich 👋 

It's possible to define nullable fields in a Prisma view. The '?' symbol is used to indicate that a field is nullable, so adding it to the Respuesta field in your view definition should make it nullable. Can you please share the query you are performing?

**Comment by flaviovich** [comment, marked as answer]
Hi @ludralph 
Here it is:
```
create view respuestasdetalle as

select u.documento_identidad DocId, u.nombre Colaborador, s.nombre Sede, p.id Pregunta, r.Respuesta,
case when p.alternativaCorrecta = r.respuesta then 'CORRECTO' else 
	case when r.respuesta is null then 'NO RESPONDIO' else 'INCORRECTO' end end Resultado
from (
	select p.id, p.alternativaCorrecta, u.id usuario_id
	from usuarios u
	cross join preguntas p
) p
left join respuestas r on p.id = r.pregunta_id and p.usuario_id = r.usuario_id
left join usuarios u on p.usuario_id = u.id
left join sedes s on u.sede_id = s.id
order by s.nombre, u.nombre, p.id
```
I think it's a Prisma/Client issue: if you declare a view with any date type error and you fix it, Prisma saved somewhere with the initial error.
It happen to me the very first time, my "solution" was to rename the view.
I hope you understand what I'm saying because I'm still learning English.
